lib/train/platforms/detect/helpers/os_common.rb

```diff
@@ -87,5 +87,65 @@ def cisco_show_version
 
       @cache[:cisco] = nil
     end
+
+    def unix_uuid
+      uuid = unix_uuid_from_chef
+      uuid = unix_uuid_from_machine_file if uuid.nil?
+      uuid = uuid_from_command if uuid.nil?
+      uuid = unix_write_uuid if uuid.nil?
+      uuid
+    end
+
+    def unix_uuid_from_chef
+      file = @backend.file('/var/chef/cache/data_collector_metadata.json')
+      if file.exist? && !file.size.zero?
+        json = ::JSON.parse(file.content)
+        return json['node_uuid'] if json['node_uuid']
+      end
+      nil
+    end
+
+    def unix_uuid_from_machine_file
+      %W(
+        /etc/machine-id
+        /var/lib/dbus/machine-id
+        /var/db/dbus/machine-id
+        /etc/machine-uuid
+        #{ENV['HOME']}/.local/etc/machine-uuid
+      ).each do |path|
+        file = @backend.file(path)
+        next unless file.exist? && !file.size.zero?
+        return file.content.chomp if path =~ /uuid/
+        return uuid_from_string(file.content.chomp)
+      end
+      nil
+    end
+
+    def unix_write_uuid
+      uuid = SecureRandom.uuid
+      result = @backend.run_command("echo \"#{uuid}\" > /etc/machine-uuid")
+      unless result.exit_status.zero?
+        # fallback to user location
+        warn 'Cannot write uuid to `/etc/machine-uuid`, falling back to ~/.local/etc/machine-uuid instead.'
+        result = @backend.run_command("mkdir -p #{ENV['HOME']}/.local/etc; echo \"#{uuid}\" > #{ENV['HOME']}/.local/etc/machine-uuid")
+        raise 'Cannot write uuid to `~/.local/etc/machine-uuid`.' unless result.exit_status.zero?
```

Comment:
Since we surround `/etc/machine-uuid` with backticks, can we do the same for `~/.local/etc/machine-uuid` as well? Also, can we interpolate this in some way so that the home directory is expanded and shown to the user?